great_expectations/agent/agent.py

```diff
@@ -212,29 +218,26 @@ def _reject_correlation_id(self, id: str):
             self._correlation_ids.clear()
         return should_reject
 
-    @classmethod
-    def _get_config(cls) -> GXAgentConfig:
+    def _get_config(self) -> GXAgentConfig:
         """Construct GXAgentConfig."""
 
         # ensure we have all required env variables, and provide a useful error if not
-
-        class GxAgentConfigSettings(pydantic.BaseSettings):
-            gx_cloud_base_url: str = "https://api.greatexpectations.io"
-            gx_cloud_organization_id: str
-            gx_cloud_access_token: str
-
         try:
-            config = GxAgentConfigSettings()
+            config = GxAgentServerSessionConfig()
+            self._agent_server_session_config = config
         except pydantic.ValidationError as validation_err:
             raise GXAgentError(
                 f"Missing or badly formed environment variable\n{validation_err.errors()}"
             ) from validation_err
 
         # obtain the broker url and queue name from Cloud
+        agent_server_session = AgentServerSession(config)
 
-        agent_sessions_url = f"{config.gx_cloud_base_url}/organizations/{config.gx_cloud_organization_id}/agent-sessions"
+        agent_sessions_url = agent_server_session.build_url(
+            resource_endpoint="/agent-sessions"
+        )
 
-        session = create_session(access_token=config.gx_cloud_access_token)
+        session = agent_server_session.session()
```

Comment:
Isn't it expensive to keep re-instantiating these Session objects? Why not do once during __init__ by instantiating a `AgentServerSession` (the non-factory version alluded to above) and sticking in `self.agent_server_session` (or do once, but using @property pattern with a `self._agent_server_session` attr). Then when needed, agent methods could do something like `self.agent_server_session.post_to_resource_endpoint(resource_endpoint, ...)`.

Reply:
I agree.
The fact that was are using a session seems like an implementation detail. `AgentServerSession` might be better named as `GxCloudClient`.

Although (non-blocking) I would also argue we don't really need this abstraction at all.

Reply:
Or maybe `AgentServerSession` should implement `__enter__` & `__exit__` so that it can be used as a context manager and the session should exist for the lifetime of the agent job.